src/shared/condition.c

```diff
@@ -578,8 +579,51 @@ static int condition_test_firmware_devicetree_compatible(const char *dtcarg) {
         return strv_contains(dtcompatlist, dtcarg);
 }
 
+static int condition_test_firmware_smbios_field(const char *expression) {
+        int r;
+        _cleanup_free_ char *field = NULL, *expected_value = NULL, *actual_value = NULL;
+        OrderOperator operator;
+
+        assert(expression);
+
+        /* Parse SMBIOS field */
+        r = extract_first_word(&expression, &field, "!<=>", EXTRACT_RETAIN_SEPARATORS);
+        if (r <= 0 || isempty(expression))
+                return -EINVAL;
+
+        /* Remove trailing spaces from SMBIOS field */
+        strstrip(field);
+
+        /* Parse operator */
+        operator = parse_order(&expression);
+        if (operator == -EINVAL)
+                return operator;
+
+        /* Parse expected value */
+        r = extract_first_word(&expression, &expected_value, NULL, EXTRACT_UNQUOTE);
+        if (r <= 0)
+                return -EINVAL;
+
+        /* Read actual value from sysfs */
+        const char *p = strjoina("/sys/class/dmi/id/", ascii_strlower(field));
+        r = read_one_line_file(p, &actual_value);
+        if (r < 0) {
+                log_debug_errno(r, "Failed to read %s: %m", p);
+                return false;
```

Comment:
please propagate unexpected errors as they are. That said, it might make sense to map ENOENT (which is kind of an expected error) more sensibly, i.e. maybe as `false`

Reply:
Adjusted in latest version